src/protocol/templates/ping1d.h.in

```diff
@@ -0,0 +1,171 @@
+     *  @code
+     *  ping_msg_ping1D_voltage_5 msg(*pd.request(Ping1DNamespace::Voltage_5));
+     *  @endcode 
+     */
+    PingMessage* request(enum Ping1DNamespace::msg_ping1D_id id, uint16_t timeout_ms = 400);
+  
+
+    /** Request a PingMessage of type T from the device
+     *  @param timeout_ms: The timeout period to wait for the requested PingMessage to be received
+     *  @return The PingMessage that was requested
+     *  @return null if the device did not reply with the requested message before the timeout period expired
+     *
+     *  @par ex.
+     *  @code
+     *  auto msg = pd.request<ping_msg_ping1D_voltage_5>();
+     *  @endcode
+     */
+    template <typename T>
+    T* request();
+
+    /** Helper to request distance and confidence
+     *  @return true if the distance and confidence have been updated successfully
+     */
+    bool update() {
+        return request(Ping1DNamespace::Distance_simple);
+    }
+
+{% for msg in messages["ping1D"]["get"] %}
+    /** {{messages["ping1D"]["get"][msg].description}}
+{% for field in messages["ping1D"]["get"][msg].payload %}
+    *   @return {{field.name}}: {% if field.units %}Units: {{field.units}}; {% endif %}{{field.description}}
+{% endfor%}
+    */
+    bool get_{{msg}}(
+{%- for field in messages["ping1D"]["get"][msg].payload %}
+{% if is_vector(field.type) %}
+{{ "                     " if not loop.first }}{{get_type_string(field.vector.sizetype)}}* {{field.name}}_length = nullptr, {{get_type_string(field.vector.datatype)}}** {{field.name}} = nullptr{{ ",\n" if not loop.last }}
+{%- else %}
+{{ "                     " if not loop.first }}{{get_type_string(field.type)}}* {{field.name}} = nullptr{{ ",\n" if not loop.last }}
+{%- endif %}
+{% endfor -%}
+         );
+
+{% endfor %}
+
+{% for msg in messages["ping1D"]["set"] %}
+    /** {{messages["ping1D"]["set"][msg].description}}
+{% for field in messages["ping1D"]["set"][msg].payload %}
+    *   @param {{field.name}} - {% if field.units %}Units: {{field.units}}; {% endif %}{{field.description}}
+{% endfor %}
+    */
+    bool {{msg}}({% for field in messages["ping1D"]["set"][msg].payload %}{{get_type_string(field.type)}} {{field.name}}, {% endfor %}bool verify = true);
+
+{% endfor %}
+
+{% for field in proplist %}
+    // Return the latest value received
+{% if is_vector(proplist[field].type) %}
+{% if proplist[field].vector.sizetype %}
+    {{get_type_string(proplist[field].vector.sizetype)}} {{proplist[field].name}}_length() { return _{{proplist[field].name}}_length; }
+    {{get_type_string(proplist[field].vector.datatype)}}* {{proplist[field].name}}() { return _{{proplist[field].name}}; }
+{% endif %}
+{% else %}
+    {{get_type_string(proplist[field].type)}} {{proplist[field].name}}() { return _{{proplist[field].name}}; }
+{% endif %}
+
+{% endfor %}
+
+private:
+    // Device I/O
+    Stream& _stream;
+
+    // For decoding PingMessages from the device
+    PingParser _parser;
+
+{% for field in proplist %}
```

Comment:
All values here should be set to zero.
```
	// Device type. 0: 1D Echosounder
    uint8_t _device_type = 0;

    // Device model. 0: Ping1D
    uint8_t _device_model = 0;

    // Firmware major version.
    uint16_t _firmware_version_major = 0;
```

Reply:
this is implicit in the language, no?

Reply:
Only for the ones created in compilation time.

Reply:
For futher info:
https://en.cppreference.com/w/cpp/language/default_initialization
*Created in compilation time*
```
    const T2 t2;      // const class, calls the user-provided default ctor
                      // t2.mem is default-initialized (to indeterminate value)
```
*Not created in compilation time*
```
int n; // static non-class, a two-phase initialization is done:
       // 1) zero initialization initializes n to zero
       // 2) default initialization does nothing, leaving n being zero
```